strings/base_array.h

```diff
@@ -418,14 +438,9 @@ WINRT_EXPORT namespace winrt
     }
 
     template <typename T>
-    auto detach_abi(com_array<T>& object) noexcept
+    std::pair<uint32_t, impl::arg_out<T>> detach_abi(com_array<T>& object) noexcept
```

Comment:
Same comment as above. This change undoes an intentional change made to address a customer-reported issue.

Reply:
Do you recall the issue? We should have a test to avoid such a regression, but all tests pass on this PR.

Reply:
Oh, you're referring to #1165 - that was itself to suppress a compiler warning. 🙄

Reply:
Ideally, we can avoid the `memset` in #1165 and just suppress the code analysis warning but I don't know whether that's possible since its presumably not an actual compiler warning. Perhaps @DefaultRyan can chime in.

Reply:
It looks to me like the original change was to suppress a warning in MSVC and this change is to suppress a warning in GCC. So if nobody objects, I'll go ahead and approve this change.